src/main/java/com/example/dog/dao/JdbcDogDAO.java

```diff
@@ -0,0 +1,127 @@
+package com.example.dog.dao;
+
+import com.example.dog.model.Dog;
+import org.springframework.jdbc.datasource.DriverManagerDataSource;
+import org.springframework.stereotype.Component;
+
+import java.sql.*;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class JdbcDogDAO implements DogDAO {
+
+    private DriverManagerDataSource dataSource;
+
+    public JdbcDogDAO(DriverManagerDataSource dataSource) {
+        this.dataSource = dataSource;
+        try (Connection conn = dataSource.getConnection()) {
+            Statement statement = conn.createStatement();
+            statement.execute("" +
+                    "CREATE TABLE DOG (" +
+                    "ID INTEGER PRIMARY KEY AUTO_INCREMENT, " +
+                    "NAME VARCHAR (255) NOT NULL, " +
+                    "HEIGHT INTEGER NOT NULL, " +
+                    "WEIGHT INTEGER NOT NULL, " +
+                    "BIRTHDAY TIMESTAMP WITH TIME ZONE)");
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public Dog get(int id) {
+        Statement statement;
+        ResultSet resultSet;
+        Dog returnedDog = null;
+        try (Connection conn = dataSource.getConnection()) {
+            statement = conn.createStatement();
+            resultSet = statement.executeQuery("select id, name, height, weight, birthday from dog where id = " + id);
+            while (resultSet.next()) {
```

Comment:
`while` or `if`? If there's more than 1 dog returned, I'd be surprised and would rather expect an error.